common/logger/src/struct_log.rs

```diff
@@ -90,6 +95,56 @@ impl StructuredLogEntry {
         ret
     }
 
+    pub fn clone_without_data(&self) -> Self {
+        Self {
+            log: self.log.clone(),
+            pattern: self.pattern,
+            category: self.category,
+            name: self.name,
+            module: self.module,
+            location: self.location,
+            git_rev: self.git_rev,
+            timestamp: self.timestamp.clone(),
+            severity: self.severity,
+            data: HashMap::new(),
+        }
+    }
+
+    fn to_json(&self) -> Result<Value, serde_json::Error> {
+        let value = serde_json::to_value(self);
+        if let Err(e) = &value {
+            log::error!("[Logging] Failed to serialize struct log entry: {}", e);
+        }
+        value
+    }
+
+    pub fn to_json_string(&self) -> Result<String, serde_json::Error> {
+        let json = self.to_json()?;
+
+        // If the message is too long, let's drop the
+        let mut json_string = json.to_string();
+        if json_string.len() > MAX_LOG_LINE_SIZE {
+            log::error!(
+                "[Logging] Structured log is too long, shrinking below : {}",
+                MAX_LOG_LINE_SIZE
+            );
+
+            let mut entry = self.clone_without_data();
+            if let Some(mut log) = entry.log {
+                log.truncate(MAX_LOG_LINE_SIZE - LOG_INFO_OFFSET);
+                // Leave 128 bytes for all the other info
+                entry.log = Some(log);
+            }
+            entry = entry.data(
```

Comment:
Do you also want to add information to data about what was original length? As a separate field so that it could be analyzed